### PR TITLE
feat: enable root span to contain route

### PR DESCRIPTION
**Pull request by DinaYakovlev**
## Which problem is this PR solving?

In addition to [#176](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/176) PR and to [#273](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/273) We should able to instrument the root span name. This way it would be clear from the beginning on what request the span is about, and if in the future there would be an option to remove layers (like in the express plugin) the route will be in the root span.

## Short description of the changes

using koa plugin before the change:
Before
using koa plugin after  the change:
After

### CODE REVIEW

File: plugins/node/opentelemetry-koa-instrumentation/src/koa.ts [hunk 1]
```diff
@@ -118,7 +121,8 @@ export class KoaInstrumentation extends BasePlugin<typeof koa> {
     middlewareLayer[kLayerPatched] = true;
     this._logger.debug('patching Koa middleware layer');
     return async (context: KoaContext, next: koa.Next) => {
-      if (api.getSpan(api.context.active()) === undefined) {
+      const parent = api.getSpan(api.context.active()) as KoaPluginSpan;
```
Comment (vmarchaud) [inline plugins/node/opentelemetry-koa-instrumentation/src/koa.ts:124]: i'm not sure why you are extending the span api, can't you use [`context()`](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-api/src/trace/span.ts#L42) instead ?

Comment (DinaYakovlev) [reply plugins/node/opentelemetry-koa-instrumentation/src/koa.ts:124]: I have to find out somehow who is the root parent span.. I do it by checking that the parentSpanId is undefined..

**Comment by shyimo** [comment]
@vmarchaud Hi !
any news regrading the review ?

File: plugins/node/opentelemetry-koa-instrumentation/src/koa.ts (hunk 1 above)
Comment (vmarchaud) [reply plugins/node/opentelemetry-koa-instrumentation/src/koa.ts:124]: I remembered that we do the same in the express plugin so thats fine

**Comment by vmarchaud** [comment]
@shyimo sorry for the delay, could you rebase the PR ?

**Comment by DinaYakovlev** [comment]
@vmarchaud rebased, please have another look :)

**Review by vmarchaud — APPROVED** [review]

File: plugins/node/opentelemetry-koa-instrumentation/src/koa.ts (hunk 1 above)
Comment (dyladan) [reply plugins/node/opentelemetry-koa-instrumentation/src/koa.ts]: This isn't safe. An instrumentation may be used by a third party SDK which doesn't have the same private properties we do. The safer (but more roundabout) way to check if it is the root would be to check if there is a span in context before the first span is created

Comment (shyimo) [reply plugins/node/opentelemetry-koa-instrumentation/src/koa.ts]: thanks @dyladan 
fixed

**Review by obecny — APPROVED** [review]

**Comment by vmarchaud** [comment]
@DinaYakovlev Thanks for the contribution and sorry for waiting so long

**Pull request merged**